docker/mac/merge_polygons_for_mac/merge_polygons.py

```diff
@@ -0,0 +1,349 @@
+import os
+import argparse
+import json
+import pandas as pd
+import geopandas as gpd
+from google.cloud import storage
+from shapely.geometry import Polygon, box
+from shapely.affinity import translate
+import matplotlib.pyplot as plt
+import warnings
+
+#BILLING_PROJECT_ID = os.environ['WORKSPACE_NAMESPACE']
+#WORKSPACE = os.environ['WORKSPACE_NAME']
+#bucket = os.environ['WORKSPACE_BUCKET']
+
+def main(cell_outlines, intervals):
+
+    cell_outlines = cell_outlines.split(",")
+
+    def list_blobs_paths(bucket, prefix):
+        """Returns a list of blob paths in the bucket that begin with the prefix."""
+        storage_client = storage.Client()
+        blobs = storage_client.list_blobs(bucket, prefix=prefix)
+        blob_paths = [blob.name for blob in blobs]
+        return blob_paths
+
+    def string_to_polygon(coord_string):
+        coords = coord_string.split(',')
+        it = iter(map(int, coords))
+        return Polygon([(x, y) for x, y in zip(it, it)])
+
+    with open(intervals, 'r') as file:
+        data_json = json.load(file)
+
+    color_list = ['red', 'blue', 'green', 'yellow', 'purple', 'black', 'orange', 'grey', 'pink', 'brown']
+
+    #bucket_name = bucket.replace('gs://', '')
+
+    #submission_id = '867d7554-6703-4f28-b144-c152a5302c70'
+    #main_workflow = '604e31c5-4b1e-4534-b1ee-4cd5af37f6ad'
+
+    gdf = None
+
+    gdf_tile = gpd.GeoDataFrame()
+
+    color_index = 0
+
+    number_of_files_to_process_start = 0
+    number_of_files_to_process_end = len(data_json['0'])
+
+    print(cell_outlines)
+    print(type(cell_outlines))
+
+    for shard_index in data_json.keys():        
+        
+        #prefix = f'submissions/{submission_id}/MAIN_WORKFLOW/{main_workflow}/call-run_cellpose_nuclear/shard-{shard_index}/cacheCopy/glob-c2c9caec5892270ef8eeb49def59c8cf/'
+        #paths = list_blobs_paths(bucket_name, prefix)
+        print(cell_outlines[number_of_files_to_process_start:number_of_files_to_process_end])
+        for inst_path in cell_outlines[number_of_files_to_process_start:number_of_files_to_process_end]:
```

Comment:
can we split this into multiple lines and double check how we are looping through these indexes?